routes/project/location/postgres.rb

```diff
@@ -215,6 +215,27 @@ class Clover
         end
       end
 
+      r.post "promote" do
+        authorize("Postgres:view", pg.id)
+
+        unless pg.read_replica?
+          if api?
+            fail CloverError.new(400, "InvalidRequest", "Non read replica servers cannot be promoted!")
+          else
+            flash["error"] = "Non read replica servers cannot be promoted!"
```

Comment:
seems like you want these two strings to be coupled (the same) but they aren't. Not sure if we use exclamation marks in our error messages either, although we don't have a style guide per se.